admin/controllers/pupil.js

```diff
@@ -40,20 +32,21 @@ controller.getAddPupil = async (req, res, next, error = null) => {
 controller.postAddPupil = async (req, res, next) => {
   res.locals.pageTitle = 'Add pupil'
   req.breadcrumbs(res.locals.pageTitle)
-  const pupilData = {
-    school: req.body.school,
-    upn: req.body.upn && req.body.upn.trim().toUpperCase(),
-    foreName: req.body.foreName,
-    lastName: req.body.lastName,
-    middleNames: req.body.middleNames,
-    gender: req.body.gender,
-    'dob-month': req.body['dob-month'],
-    'dob-day': req.body['dob-day'],
-    'dob-year': req.body['dob-year'],
-    pin: null,
-    pinExpired: false
-  }
   try {
+    const school = await schoolDataService.sqlFindOneByDfeNumber(req.user.School)
+    const pupilData = {
+      school: school.dfeNumber, // Change to `id` when saving to SQL Server
+      upn: req.body.upn && req.body.upn.trim().toUpperCase(),
```

Comment:
do we force upperCase in the service too?  feels like a good place to catch it...

Reply:
Good catch!  I've moved this into the service